### PR TITLE
[Fix] Fix some bugs in `Det3DLocalVisualizer` and update docstring.

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/visualization/local_visualizer.py [hunk 1]
```diff
@@ -43,31 +44,31 @@ class Det3DLocalVisualizer(DetLocalVisualizer):
 
     Args:
         name (str): Name of the instance. Defaults to 'visualizer'.
-        points (numpy.array, shape=[N, 3+C]): points to visualize.
-        image (np.ndarray, optional): the origin image to draw. The format
+        points (np.ndarray, optional, shape=[N, 3+C]): Points to visualize.
```
Comment (ZCMax) [inline mmdet3d/visualization/local_visualizer.py:47]: Points to visualize with shape (N, 3+C)

File: mmdet3d/visualization/local_visualizer.py [hunk 2]
```diff
@@ -152,32 +152,33 @@ def set_points(self,
                    pcd_mode: int = 0,
                    vis_mode: str = 'replace',
                    frame_cfg: dict = dict(size=1, origin=[0, 0, 0]),
-                   points_color: Tuple = (0.5, 0.5, 0.5),
+                   points_color: Tuple[float] = (0.5, 0.5, 0.5),
                    points_size: int = 2,
                    mode: str = 'xyz') -> None:
-        """Set the points to draw.
+        """Set the pints to draw.
```
Comment (ZCMax) [inline mmdet3d/visualization/local_visualizer.py:158]: points

File: mmdet3d/visualization/local_visualizer.py [hunk 3]
```diff
@@ -540,21 +539,26 @@ def draw_seg_mask(self, seg_mask_colors: np.array):
...
-            vis_task (str): Visualiztion task, it includes:
+            input_meta (dict): Meta information.
+            vis_task (str): Visualization task, it includes:
                 'lidar_det', 'multi-modality_det', 'mono_det'.
+            palette (List[tuple], optional): Palette information
+                corresponding to the category. Defaults to None.
 
         Returns:
-            dict: the drawn point cloud and image which channel is RGB.
+            dict: The drawn point cloud and image which channel is RGB.
```
Comment (ZCMax) [inline mmdet3d/visualization/local_visualizer.py:561]: which -> whose

File: mmdet3d/visualization/local_visualizer.py [hunk 4]
```diff
@@ -660,35 +654,36 @@ def show(self,
...
     def add_datasample(self,
                        name: str,
                        data_input: dict,
-                       data_sample: Optional['Det3DDataSample'] = None,
+                       data_sample: Optional[Det3DDataSample] = None,
                        draw_gt: bool = True,
                        draw_pred: bool = True,
                        show: bool = False,
-                       wait_time: float = 0,
+                       wait_time: int = 0,
```
Comment (ZwwWayne) [inline mmdet3d/visualization/local_visualizer.py:672]: seems the type is changed here, so can we use float?

File: mmdet3d/visualization/local_visualizer.py [hunk 5]
```diff
@@ -598,21 +595,16 @@ def _draw_pts_sem_seg(self,
                           points: Union[Tensor, np.ndarray],
                           pts_seg: PointData,
                           palette: Optional[List[tuple]] = None,
-                          ignore_index: Optional[int] = None):
+                          ignore_index: Optional[int] = None) -> None:
```
Comment (JingweiZhang12) [inline mmdet3d/visualization/local_visualizer.py:598]: Please make sure that the visualization of the `lidar_seg` task can run successfully.

Comment (Xiangxu-0103) [reply mmdet3d/visualization/local_visualizer.py:608]: Have checked.

File: mmdet3d/visualization/local_visualizer.py [hunk 6]
```diff
@@ -723,9 +731,10 @@ def add_datasample(self,
 
         if draw_gt and data_sample is not None:
             if 'gt_instances_3d' in data_sample:
-                gt_data_3d = self._draw_instances_3d(
-                    data_input, data_sample.gt_instances_3d,
-                    data_sample.metainfo, vis_task, palette)
+                if len(data_sample.gt_instances_3d) > 0:
```
Comment (ZwwWayne) [inline mmdet3d/visualization/local_visualizer.py:734]: This has been fixed in https://github.com/open-mmlab/mmdetection3d/pull/2231, we do not need to modify this part

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**